File: go-deps/go-deps.go
package main

import (
   "bufio"
   "bytes"
   "github.com/89z/x"
   "log"
   "os"
   "os/exec"
   "strings"
)

func check(e error) {
   if e != nil {
      log.Fatal(e)
   }
}

func main() {
   if len(os.Args) != 2 {
      println(`usage:
go-deps <URL>

example:
   go-deps https://github.com/dinedal/textql`)
      os.Exit(1)
   }
   mod := os.Args[1][8:]
   e := x.System("go", "mod", "init", "deps")
   check(e)
   e = x.System("go", "get", mod)
   check(e)
   dep, e := x.Popen("go", "list", "-deps", mod + "/...")
   check(e)
   os.Remove("go.mod")
   os.Remove("go.sum")
   deps := 0
   for dep.Scan() {
      text := dep.Text()
      if strings.Contains(text, "/internal/") {
         continue
      }
      if ! strings.Contains(text, ".") {
         continue
      }
      if strings.HasPrefix(text, mod + "/") {
         continue
      }
      println(text)
      deps++
   }
   print("\n", deps, " deps\n")
}

File: go-deps/go-deps.php
<?php
declare(strict_types = 1);

if ($argc != 2) {
   echo <<<eof
usage:
   go-deps.php <URL>

example:
   go-deps.php https://github.com/dinedal/textql
eof;
   exit(1);
}

$url_s = $argv[1];
$url_m = parse_url($url_s);
$mod_s = $url_m['host'] . $url_m['path'];
system('go mod init deps');
system('go get ' . $mod_s);
exec('go list -deps ' . $mod_s . '/...', $dep_a);
unlink('go.sum');
unlink('go.mod');
$dep_n = 0;

foreach ($dep_a as $dep_s) {
   if (str_contains($dep_s, '/internal/')) {
      continue;
   }
   if (! str_contains($dep_s, '.')) {
      continue;
   }
   if (str_starts_with($dep_s, $mod_s . '/')) {
      continue;
   }
   echo $dep_s, "\n";
   $dep_n++;
}

echo "\n", $dep_n, " deps\n";
